homeassistant/helpers/temperature.py

```diff
@@ -3,13 +3,13 @@
 
 from numbers import Number
 
-from homeassistant.const import PRECISION_HALVES, PRECISION_TENTHS
+from homeassistant.const import Precision
 from homeassistant.core import HomeAssistant
 from homeassistant.util.unit_conversion import TemperatureConverter
 
 
 def display_temp(
-    hass: HomeAssistant, temperature: float | None, unit: str, precision: float
+    hass: HomeAssistant, temperature: float | None, unit: str, precision: Precision
```

Comment:
This sure looks like it will break the climate platform as its currently written